src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/AuditUtility.cs

```diff
@@ -257,11 +258,11 @@ private static (string severityLabel, NuGetLogCode code) GetSeverityLabelAndCode
 
             foreach (RestoreTargetGraph graph in _targetGraphs)
             {
-                LibraryIdentity? currentProject = graph.Graphs.FirstOrDefault()?.Item.Key;
+                GraphItem<RemoteResolveResult>? currentProject = graph.Graphs.FirstOrDefault()?.Item;
 
-                foreach (ResolvedDependencyKey resolvedDependency in graph.ResolvedDependencies.Where(dep => dep.Child.Type == LibraryType.Package))
+                foreach (GraphItem<RemoteResolveResult>? node in graph.Flattened.Where(r => r.Key.Type == LibraryType.Package))
```

Comment:
Just for my own learning, why Flattened and ResolvedDependencies wouldn't have the same nodes?

Reply:
Consider the first example from the docs page's "cousin dependencies" example: https://learn.microsoft.com/en-us/nuget/concepts/dependency-resolution#cousin-dependencies

`ResolvedDependeies` shows all the graph edges, including the one that says "PackageA version 1.0.0 depends on PackageB 1.0.0", even though PackageB version 2.0.0 was selected.

On the other hand, `Flattened` only contains the selected nodes (not edges)